### PR TITLE
XAI cls notebook update

### PR DESCRIPTION
**Pull request by GalyaZalesskaya**
What was done:

- Updated structure according to Mark and Minje
image
- Updated calls according to updated API
- Updated comments & text
- Fixed overlay issue

Issues:
-  `explanation = explainer(image, target_explain_group=TargetExplainGroup.ALL)` fails with `error: OpenCV(4.9.0) /io/opencv/modules/imgproc/src/resize.cpp:3789: error: (-215:Assertion failed) !dsize.empty() in function 'resize'`

TODO:
- Update README: update structure, add reciprocam arxiv link
- Substitute loading `tests/assets/cheetah_person.jpg` image with wget link (can be done during PR to OV_notebooks)
- Fix minor text issues
- Update pypl modules for numpy 2.0 to compile
- Check that `target_layer` for MobiilenetV3 is correct. It returns correct values, but it's not the last conv layer

Question:
- Do we need `Advanced section` and `Insert XAI branch` section
- There's duplicated code 
```
# Check saved saliency maps
print(f"Saliency maps were generated for the following classes: {list(explanation.saliency_map.keys())}")
print(f"Saliency map size: {explanation.shape}")

# Show saliency maps for cheetah class
cheetah_raw_sal_map = explanation.saliency_map[cheetah_class_index]
cheetah_raw_sal_map = cv2.cvtColor(cheetah_raw_sal_map, cv2.COLOR_BGR2RGB)

plt.imshow(cheetah_raw_sal_map)
plt.show()
```
Can be moved to xai.utils as `get_sal_map_info` for better readability

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -10,8 +10,8 @@ name = "openvino_xai"
 version = "1.0.0rc01"
 dependencies = [
   "openvino-dev==2023.2",
-  "opencv-python",
-  "numpy",
+  "opencv-python<4.10",
+  "numpy<2",
```
Comment (GalyaZalesskaya) [inline pyproject.toml:14]: Temporal changes to avoid numpy compiling issue

**Comment by GalyaZalesskaya** [comment]
@itrushkin Hi Ilya, there's a draft version of notebook that we discussed. Will appreciate any feedback. Also, there's a question: is there a preferable for OV_notebooks tool to generate `Table of Content` and add "to the top" links in esch cell as it's done in OV_notebooks?

**Comment by goodsong81** [comment]
Merged to other PR

**Comment by itrushkin** [comment]
> @itrushkin Hi Ilya, there's a draft version of notebook that we discussed. Will appreciate any feedback. Also, there's a question: is there a preferable for OV_notebooks tool to generate `Table of Content` and add "to the top" links in esch cell as it's done in OV_notebooks?

There is a Python script that generates a table of contents and links to the top using a headers structure: https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/.ci/table_of_content.py
It expects `-s` argument for the notebook location.

**Pull request closed without merging**